### PR TITLE
thermostat min/max temperature clarification

### PR DESCRIPTION
**Pull request by larsonmpdx**
## Description:

**Related issue (if applicable):** see discussion in https://github.com/esphome/issues/issues/1507

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by glmnet** [comment]
@kbx81 could you review this?

**Review by kbx81 — APPROVED** [review]
LGTM

File: components/climate/thermostat.rst [hunk 1]
```diff
@@ -153,6 +153,9 @@ These temperatures are used when the device first starts up.
 **At least one of** ``default_target_temperature_low`` **and** ``default_target_temperature_high``
 **must be specified.**
 
+``min_temperature`` and ``max_temperature`` from the base climate component **will limit the range**
+of allowed temperature values in the thermostat component.  See :doc:`/components/climate/index`
+
```
Comment (kbx81) [inline components/climate/thermostat.rst:158]: I guess the only additional comment I would have is to perhaps add a sentence to address the people who ask "why"? For example:
_"These values are used to ensure any given set point is **always** within the desired range -- you wouldn't  want a broken automation in HA to set your living room's thermostat to -999999 degrees in the middle of winter, would you?"_

Totally optional and absolutely not necessary...but...we're engineers...and inevitably people jump on the discord server and ask "why?"

Comment (larsonmpdx) [reply components/climate/thermostat.rst:158]: thanks see update

File: components/climate/thermostat.rst [hunk 2]
```diff
@@ -153,6 +153,9 @@ These temperatures are used when the device first starts up.
 **At least one of** ``default_target_temperature_low`` **and** ``default_target_temperature_high``
 **must be specified.**
 
+``min_temperature`` and ``max_temperature`` from the base climate component **are used to define the range**
+of allowed temperature values in the thermostat component.  See :doc:`/components/climate/index`
```
Comment (frenck) [inline components/climate/thermostat.rst]: Use a single space after periods :)

```suggestion
of allowed temperature values in the thermostat component. See :doc:`/components/climate/index`
```

See also: <https://docs.microsoft.com/en-us/style-guide/punctuation/periods>

File: components/climate/thermostat.rst [hunk 3]
```diff
@@ -268,7 +271,9 @@ Advanced Options
 .. note::
 
     While this platform uses the term temperature everywhere, it can also be used to regulate other values.
-    For example, controlling humidity is also possible with this platform.
+    For example, controlling humidity is also possible with this platform.  Note that ``min_temperature`` and
+    ``max_temperature`` from the base climate component are used the define the range of adjustability and the
+    defaults will probably not make sense for control of things like humidity.  See :doc:`/components/climate/index`
```
Comment (frenck) [inline components/climate/thermostat.rst]:
```suggestion
    For example, controlling humidity is also possible with this platform. Note that ``min_temperature`` and
    ``max_temperature`` from the base climate component are used the define the range of adjustability and the
    defaults will probably not make sense for control of things like humidity. See :doc:`/components/climate/index`
```

**Review by frenck — APPROVED** [review]
Thanks, @larsonmpdx 👍 

../Frenck 🚂

**Pull request merged**